pkg/services/signingkeys/signingkeysimpl/service.go

```diff
@@ -0,0 +1,90 @@
+package signingkeysimpl
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/go-jose/go-jose/v3"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/signingkeys"
+)
+
+const (
+	serverPrivateKeyID = "default"
+)
+
+var _ signingkeys.Service = new(Service)
+
+func ProvideEmbeddedSigningKeysService() *Service {
+	s := &Service{
+		log: log.New("auth.key_service"),
+	}
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
```

Comment:
Didn't you want to use ECDSA?  What made you change your mind?

Reply:
It can be added later, the goal is to satisfy the requirements of the oauth2 server poc.

Reply:
Alright :)